### ISSUE TITLE
Build fails with newer mupdf, api changes

### ISSUE THREAD

**Issue by wltjr** [vtorri/etui#28]
Seems like mupdf changed with 1.12.0 from 1.11. I believe that is where the following error is coming from

```c
src/modules/pdf/etui_module_pdf.c:871:13: error: too few arguments to function  fz_new_pixmap_with_bbox_and_data?
     image = fz_new_pixmap_with_bbox_and_data(md->doc.ctx,
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I will see about fixing time permitting. Along with the other issue regarding theme I opened. I will definitely do something on that one. Just mentioning this in case you get a chance to fix, or at least aware of the issue.

**Comment by vtorri** [comment]
mupdf devs are breaking api each time they provide a new release
in addition, there is no easy way to know if the version is 1.11 or 1.12. The version is in fitz/version.h and it is a string (arg...)
what I can do is adding a check in configure to select the version but the link against mupdf depends on the distro...
i can also add a configure option that the user will have to pass to specify mupdf version, but then it's not automatic

if you have a better solution to know, at preprocessor time, the mupdf version, i would gladly fix that

Note that most current linux distro are providing mupdf 1.11

**Comment by wltjr** [comment]
Wonderful they do such things... I can remain with 1.11. Maybe we need to get in touch with them, and request as least some means to determine version number. Along with request to not break API like such. Seems its just 1 function, so not to much to request they keep a legacy wrapper or something for 1-2 versions. Clearly not an ideal upstream. No worries, we can address in time. Least its known.

I would avoid having to specify version, unless it was something that came from pkg-config. Which would be automatic, and seems that is not an option. Not sure as to solution I will have to take a look. Thank  you for the information, and limitations.

Seems [1.12 came out mid December](https://mupdf.com/downloads/). Not sure on distro adoption, but seems like they would be looking to add that to unstable or something. Either way I can see about poking at this time permitting. Unless you get to it first.

**Comment by vtorri** [comment]
i already told them to modify their build system to create shared library (atleast with dlopen, we could do some tests...) and to add pkgconfig files some years ago. Their answer is 'never'...

**Comment by wltjr** [comment]
Interesting, maybe their intentions is for their own usage and not for others to use as a library. Probably so they do not have to worry about API changes etc. Though it usually is not much. With my [jem, java env manager](https://github.com/Obsidian-StudiosInc/jem) I went ahead and made a .so for a variety of uses. Even though its basically a cli program. It may be used from python or other. CLI is just a wrapper around the shared library. Anyway maybe I will see about another request from them. Otherwise will need to get creating or something.

**Comment by NuLogicSystems** [comment]
I'm getting the same errors on Arch/Manjaro,

**Comment by vtorri** [comment]
fixed in commit 0021e292e6010232e2f2d870e18783d0e8b65ec3
